cub/cub/util_device.cuh

```diff
@@ -635,18 +636,79 @@ struct ChainedPolicy
   template <typename FunctorT>
   CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t Invoke(int device_ptx_version, FunctorT& op)
   {
+    // __CUDA_ARCH_LIST__ is only available from CTK 11.5 onwards
+#ifdef __CUDA_ARCH_LIST__
```

Comment:
suggested: as described below, this is safe only when we have namespace magic. Although this leads to UB when linking TUs compiled for different architecture sets, old chained policy would at least find a policy to get executed. New chained policy would not. So, I'd prefer something along the lines of:

```suggestion
#if defined(__CUDA_ARCH_LIST__) && !defined(CUB_DISABLE_NAMESPACE_MAGIC)
```

but rapids is going to disable namespace magic, so a change like that would make this PR less useful. I guess if we change `cudaError_t e = cudaSuccess;` to return an actual error as suggested below, we'll at least catch the problem at runtime. I'd recommend adding a note close to the error we are going to return when no arch from `CudaArches` matched device ptx suggesting users to re-enable namespace magic or wrap namespace.

Reply:
> I'd recommend adding a note close to the error we are going to return when no arch from CudaArches matched device ptx suggesting users to re-enable namespace magic or wrap namespace.

I've added a comment on the error. Could you take a look and let me know if it's indicative?